packages/dds/tree/src/forest/types/reader.ts

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Serializable } from "@fluidframework/datastore-definitions";
+import { TreeKey, TreeType } from "../..";
+
+export const enum TreeReadResult {
+    NotFound = -1,
+    Pending = 0,
+    Ok = 1,
+}
+
+export interface INodeReader {
```

Comment:
I've always found the type names "reader" confusing.

Is the app logic that want wants to read the tree the reader, of is the API calls to the reading the reader? This ambiguity makes me tend to avoid the name "reader".

In this specific case, I'd be tempted to call  INodeReader oen of ForestTree, TreeView, ForestTreeView, IterableTree
ITreeReader -> one of ForestIterator, TreeIterator, TreeCursor, Forest Cursor etc.

Reply:
For the moment, I'll combine the interfaces as ITreeCursor.

I may split them again when we introduce 'visit()', which is where I passed an 'INodeReader' that was not an ITreeReader.

Reply:
INodeReader and ITreeReader were previously split to support passing the current cursor instance to a visitor without encouraging the visitor the to move the cursor themselves.

I haven't benchmarked visit() to see if it's advantageous, so I'll omit it for now and combine the interfaces into ITreeCursor.